nextcord/application_command.py

```diff
@@ -3276,3 +3293,183 @@ def unpack_annotation(
         raise ValueError(f"Given Annotation {given_annotation} has an unhandled origin: {origin}")
 
     return type_ret, literal_ret
+
+
+class Range:
+    """An annotation helper for defining slash command `min_value` and `max_value` parameters.
+
+    .. versionadded:: 2.2
+
+    .. container:: operations
+
+        .. describe:: Range[x, y]
+
+            Creates a range from ``x`` to ``y``.
+
+        .. describe:: Range[x] | Range[..., x]
+
+            Create a range up to ``x``.
+
+        .. describe:: Range[x, ...]
+
+            Create a range from ``x``.
+    """
+
+    min: ClassVar[Optional[Union[int, float]]]
+    max: ClassVar[Optional[Union[int, float]]]
+
+    @overload
+    def __class_getitem__(
+        cls,
+        value: Union[
+            int,
+            Tuple[int, int],
+            Tuple[int, EllipsisType],
+            Tuple[EllipsisType, int],
+        ],
+    ) -> Type[int]:
+        ...
+
+    @overload
+    def __class_getitem__(
+        cls,
+        value: Union[
+            float,
+            Tuple[float, float],
+            Tuple[float, EllipsisType],
+            Tuple[EllipsisType, float],
+        ],
+    ) -> Type[float]:
+        ...
+
+    def __class_getitem__(
+        cls,
+        value: Union[
+            int,
+            float,
+            Tuple[int, int],
+            Tuple[float, float],
+            Tuple[Union[int, float], EllipsisType],
+            Tuple[EllipsisType, Union[int, float]],
+        ],
+    ) -> Type[Union[int, float]]:
+        class Inner(Range, OptionConverter):
+            def __init__(self):
+                super().__init__(option_type=type(self.min or self.max))
+
+            async def convert(self, interaction: Interaction, value: Any) -> Any:
+                return value
+
+            def modify(self, option: SlashCommandOption):
+                if self.min:
+                    if option.min_value is None:
+                        option.min_value = self.min
+                if self.max:
+                    if option.max_value is None:
+                        option.max_value = self.max
```

Comment:
Can do with less nesting
```suggestion
                if self.min and option.min_value is None:
                    option.min_value = self.min
                if self.max and option.max_value is None:
                    option.max_value = self.max
```